test/jdk/java/awt/print/PrinterJob/ValidatePage/ValidatePage.java

```diff
@@ -21,407 +21,293 @@
+            return ds;
+        }
     }
 
-
-   public static void printInstructions( String[] instructions )
-    {
-      dialog.printInstructions( instructions );
+    public ValidatePage() {
+        super("Validate Page Test");
+        myPrinterJob = PrinterJob.getPrinterJob();
+        myPageFormat = new PageFormat();
+        Paper p = new Paper();
+        p.setSize(28 * 72, 21.5 * 72);
+        myPageFormat.setPaper(p);
+        setLayout(new FlowLayout());
+        Panel pfp = new Panel();
+        pfp.setLayout(new GridLayout(9, 1, 0, 0));
+        pfp.add(myOrientationLabel = new Label());
+        pfp.add(myWidthLabel = new Label());
+        pfp.add(myImageableXLabel = new Label());
+        pfp.add(myImageableRightLabel = new Label());
+        pfp.add(myImageableWidthLabel = new Label());
+        pfp.add(myHeightLabel = new Label());
+        pfp.add(myImageableYLabel = new Label());
+        pfp.add(myImageableBottomLabel = new Label());
+        pfp.add(myImageableHeightLabel = new Label());
+
+        add(pfp);
+
+        Panel pp = new Panel();
+        pp.setLayout(new GridLayout(8, 1, 0, 0));
+        pp.add(pw = new Label());
+        pp.add(pglm = new Label());
+        pp.add(pgtm = new Label());
+        pp.add(ph = new Label());
+        pp.add(pgiw = new Label());
+        pp.add(pgih = new Label());
+        pp.add(pgrm = new Label());
+        pp.add(pgbm = new Label());
+
+        add(pp);
+
+        Panel epp = new Panel();
+        epp.setLayout(new GridLayout(6, 2, 0, 0));
+
+        epp.add(new Label("Page width:"));
+        epp.add(tpw = new TextField());
+        epp.add(new Label("Page height:"));
+        epp.add(tph = new TextField());
+        epp.add(new Label("Left Margin:"));
+        epp.add(tpglm = new TextField());
+        epp.add(new Label("Top margin:"));
+        epp.add(tpgtm = new TextField());
+        epp.add(new Label("Imageable Wid:"));
+        epp.add(tpgiw = new TextField());
+        epp.add(new Label("Imageable Hgt:"));
+        epp.add(tpgih = new TextField());
+
+        add(epp);
+        displayPageFormatAttributes();
+
+        Panel panel = new Panel();
+        Button defButton = new Button("Default Page");
+        defButton.addActionListener(e -> {
+            myPageFormat = myPrinterJob.defaultPage();
+            displayPageFormatAttributes();
+        });
+
+        Button pageButton = new Button("Page Setup..");
+        pageButton.addActionListener(e -> {
+            myPageFormat = myPrinterJob.pageDialog(myPageFormat);
+            displayPageFormatAttributes();
+        });
+        Button printButton = new Button("Print");
+        printButton.addActionListener(e -> {
+            try {
+                myPrinterJob.setPrintable(ValidatePage.this, myPageFormat);
+                myPrinterJob.print();
+            } catch (PrinterException pe) {
+                PassFailJFrame.forceFail( "Test Failed");
+                pe.printStackTrace();
+            }
```

Comment:
Provide a better description of the failure, at least mention `PrinterException` was caught.

Print the stack trace before failing the test to ensure it's printed.